### PR TITLE
[onert] Introduce NUM_THREADS config

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introduces `NUM_THREADS`.
`NUM_THREADS` replaces `RUY_THREADS` and `XNNPACK_THREADS`.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: https://github.com/Samsung/ONE/issues/13909#issuecomment-2327982517

### CODE REVIEW

**Review by jyoungyun — DISMISSED** [review]
LGTM

File: runtime/onert/core/include/util/Config.lst [hunk 1]
```diff
@@ -31,8 +31,7 @@ CONFIG(USE_SCHEDULER           , bool         , "0")
 CONFIG(TRACING_MODE            , bool         , "0")
 CONFIG(MINMAX_DUMP             , bool         , "0")
 CONFIG(FP16_ENABLE             , bool         , "0")
-CONFIG(RUY_THREADS             , int          , "-1")
-CONFIG(XNNPACK_THREADS         , int          , "-1")
+CONFIG(ONERT_THREADS           , int          , "-1")
```
Comment (glistening) [inline runtime/onert/core/include/util/Config.lst:34]: (optional) What about removing `ONERT_` as other variables? We may use `NUM_THREADS`.
```suggestion
CONFIG(NUM_THREADS           , int          , "-1")
```
It seems our python code also uses the name of `num_threads`.

**Review by glistening — APPROVED** [review]
LGTM though I prefer `NUM_THREADS`. See https://github.com/Samsung/ONE/pull/13929#discussion_r1743207894.

**Review by glistening — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]
LGTM

**Pull request merged**